core/src/main/python/synapse/ml/core/logging/SynapseMLLogger.py

```diff
@@ -213,6 +218,58 @@ def log_decorator_wrapper(self, *args, **kwargs):
 
         return get_wrapper
 
+    @staticmethod
+    def log_verb_static(method_name: Optional[str] = None,  feature_name: Optional[str] = None, custom_log_function = None):
+        def get_wrapper(func):
+            @functools.wraps(func)
+            def log_decorator_wrapper(self, *args, **kwargs):
+                # Validate that self.logger is set
+                if not hasattr(self, "logger"):
```

Comment:
If it has `logger` as an atttribute, this becomes part of the public API. I suggest something like `_logger`. Or even make it configurable like the method name is configurable.

Reply:
I can make the change to `_logger` for now